src/jit/codegencommon.cpp

```diff
@@ -2071,9 +2071,13 @@ void CodeGen::genGenerateCode(void** codePtr, ULONG* nativeSizeOfCode)
         {
             printf("; Tier-0 compilation\n");
         }
-        if (compiler->opts.jitFlags->IsSet(JitFlags::JIT_FLAG_TIER1))
+        else if (compiler->opts.jitFlags->IsSet(JitFlags::JIT_FLAG_TIER1))
         {
-            printf("; Tier-1 compilation\n");
+            printf("; Tier-1 recompilation\n");
```

Comment:
It may be the first compilation with the tier 1 flag and may not be a recompilation. Currently the only case is for a method attributed with `MethodImplOptions.AggressiveOptimization`, such that if tier 1 JIT were to diverge and do more optimizations those methods could benefit from them.
```suggestion
            printf("; Tier-1 compilation\n");
```